File: 703092-two-sum-pair-with-given-sum-GFG/703092-two-sum-pair-with-given-sum.cpp
class Solution {
  public:
    bool twoSum(vector<int>& arr, int target) {
        sort(arr.begin(),arr.end());
        int left =0;
        int right = arr.size()-1;
        while(left<right){
            
            int sum = arr[left]+arr[right];
            
            if(sum == target) return true;
            
            if(sum > target) right--;
            else left++;
        }
        return false;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 703092-two-sum-pair-with-given-sum-GFG/703092-two-sum-pair-with-given-sum.py
class Solution:
	def twoSum(self, arr, target):
		seen = {}
        
        for i,x in enumerate(arr):
            req = target-x
            
            if req in seen:
                return True
            
            seen[x] = i
        
        return False    

# Synced seamlessly with LeetHub Pro
# Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
# Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna